apps/web/vibes/soul/sections/dynamic-form-section/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { DynamicForm, DynamicFormAction } from '@/vibes/soul/primitives/dynamic-form';
+import { Field, FieldGroup } from '@/vibes/soul/primitives/dynamic-form/schema';
+import { SectionLayout } from '@/vibes/soul/sections/section-layout';
+
+interface Props<F extends Field> {
+  title?: string;
+  subtitle?: string;
+  action: DynamicFormAction<F>;
+  fields: Array<F | FieldGroup<F>>;
+  submitLabel?: string;
+  className?: string;
+}
+
+export function DynamicFormSection<F extends Field>({
+  className,
+  title,
+  subtitle,
+  fields,
+  submitLabel,
+  action,
+}: Props<F>) {
+  return (
+    <SectionLayout className={className} containerSize="lg">
+      {Boolean(title) && (
+        <header className="mx-auto w-full max-w-4xl pb-8 @2xl:pb-12 @4xl:pb-16">
+          <h1 className="mb-5 font-heading text-4xl font-medium leading-none @xl:text-5xl">
+            {title}
+          </h1>
+          {Boolean(subtitle) && subtitle !== '' && (
```

Comment:
I think we should, as a general rule, avoid coercing values using constructors of primitives, like `Boolean`. Instead, we should be explicit and check `subtitle != null` which will cover `null` and `undefined`.

Reply:
This also goes for the usage for `title` above

Reply:
@migueloller Hmm, that is an interesting approach. I've always went by the opposite, that strict equal `===` `!==` should always be preferred over the simple equal. Is there a reason why we would prefer that over type coercion with the `Boolean` constructor? The Boolean constructor seems like a better way to capture what we intend with less code. In fact, using `Boolean(subtitle)` technically means the `subtitle !== ''` is also unnecessary:

> `Boolean(null) = false`
> `Boolean(undefined) = false`
> `Boolean('') = false`

Reply:
If the intent is to handle falsy values like `''`, then yes, I agree with you that it is more succinct than having to do `subtitle != null & subtitle !== ''`. That's good w/ me. As for always preferring `===` over `==`, that's a rule that I always follow, except when checking for both `null` and `undefined`. Instead of `val !== null && val !== undefined`, I opt for `val != null` since that's equivalent. I think the same logic applies to `Boolean(val)`. In that case, though, you could also do `!val` or `!!val`. I guess `Boolean(val)` feels more "explicit" and we can reserve `!` only for boolean values.